test/tools/cli/IonProcess.java

```diff
@@ -61,66 +62,68 @@ public static void main(final String[] args) {
-                    ) {
-
-                currentInfo.setFileName(path);
+            ) {
+                processContext.setFileName(path);
 
                 if (isEventStream(ionReader)) {
-                    currentInfo.setEventIndex(currentInfo.getEventIndex() - 1);
-                    processFromEventStream(ionWriterForOutput, ionWriterForErrorReport, ionReader, currentInfo, args);
+                    processContext.setEventIndex(processContext.getEventIndex() - 1);
+                    processFromEventStream(ionWriterForOutput, ionWriterForErrorReport,
+                            ionReader, processContext, args, outputStream);
                 } else {
-                    processFromIonStream(ionWriterForOutput, ionWriterForErrorReport, ionReader, currentInfo, args);
+                    processFromIonStream(ionWriterForOutput, ionWriterForErrorReport, ionReader, processContext, args);
                 }
 
                 ionWriterForOutput.finish();
                 ionWriterForErrorReport.finish();
             } catch (IonException e) {
-                if (args.getOutputFormat() == OutputFormat.EVENTS) {
-                    new ErrorDescription(ErrorType.READ, e.getMessage(), currentInfo.getFileName(),
-                            currentInfo.getEventIndex()).writeOutput(ionWriterForErrorReport);
+                if (args.getOutputFormat() == OutputFormat.EVENTS || processContext.getState() == ErrorType.WRITE) {
+                    new ErrorDescription(processContext.getState(), e.getMessage(), processContext.getFileName(),
+                            processContext.getEventIndex()).writeOutput(ionWriterForErrorReport);
                 } else {
-                    new ErrorDescription(ErrorType.READ, e.getMessage(), currentInfo.getFileName())
+                    new ErrorDescription(processContext.getState(), e.getMessage(), processContext.getFileName())
                             .writeOutput(ionWriterForErrorReport);
                 }
                 System.exit(IO_ERROR_EXIT_CODE);
-            }  catch (IllegalStateException e) {
+            } catch (Exception e) {
                 if (args.getOutputFormat() == OutputFormat.EVENTS) {
-                    new ErrorDescription(ErrorType.STATE, e.getMessage(), currentInfo.getFileName(),
-                            currentInfo.getEventIndex()).writeOutput(ionWriterForErrorReport);
+                    new ErrorDescription(ErrorType.STATE, e.getMessage(), processContext.getFileName(),
+                            processContext.getEventIndex()).writeOutput(ionWriterForErrorReport);
                 } else {
-                    new ErrorDescription(ErrorType.STATE, e.getMessage(), currentInfo.getFileName())
+                    new ErrorDescription(ErrorType.STATE, e.getMessage(), processContext.getFileName())
                             .writeOutput(ionWriterForErrorReport);
                 }
```

Comment:
Same question about unifying the branches.

Reply:
Changed